src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -1388,6 +1404,21 @@ protected override void DecodeWellKnownAttributeImpl(ref DecodeWellKnownAttribut
                     diagnostics.Add(ErrorCode.ERR_UnscopedRefAttributeUnsupportedMemberTarget, arguments.AttributeSyntaxOpt.Location);
                 }
             }
+            else if (attribute.IsTargetAttribute(AttributeDescription.OverloadResolutionPriorityAttribute))
+            {
+                MessageID.IDS_OverloadResolutionPriority.CheckFeatureAvailability(diagnostics, arguments.AttributeSyntaxOpt);
+
+                if (!IsIndexer || IsExplicitInterfaceImplementation)
```

Comment:
>IsExplicitInterfaceImplementation [](http://example.com/codeflow?start=34&length=33)

Should the `IsExplicitInterfaceImplementation` check be in `CanHaveOverloadResolutionPriority`?
Then we can have the same basic logic here: `if (!CanHaveOverloadResolutionPriority) { ... report diagnostic with error code 1 or 2 ... }`

Reply:
What was the resolution?

Reply:
I did exactly what you suggested

Reply:
The second part of my suggestion was to move the `IsOverride` case into the `!CanHaveOverloadResolutionPriority` case. If it makes sense, one form could be:
```
if (!CanHaveOverloadResolutionPriority)
{
  diagnostics.Add(IsOverride ? ErrorCode.ERR_CannotApplyOverloadResolutionPriorityToOverride : ErrorCode.ERR_CannotApplyOverloadResolutionPriorityToMember, arguments.AttributeSyntaxOpt.Location);
}
```